### ISSUE TITLE
Add ability to execute "builtin" ethereum functions

### ISSUE THREAD

**Issue by armaniferrante** [armaniferrante/pure-evm#7]
Builtin functions like `sha256` and `recover` execute as cross contract calls to fixed account addresses on Ethereum.

Currently, pure-evm doesn't allow any cross contract calls whatsoever. 

It might be desireable to special case "builtin" functions as requested by https://github.com/armaniferrante/pure-evm/issues/6.

**Comment by ArjunBhuptani** [comment]
If we were to implement this ourselves and PR it, how would this work?

**Comment by ArjunBhuptani** [comment]
So all we should need to do is to implement the following:
```
call(...) {
    fn call(
    &mut self,
    _gas: &U256,
    _sender_address: &Address,
    _receive_address: &Address,
    _value: Option<U256>,
    _data: &[u8],
    _code_address: &Address,
    _call_type: CallType,
    _trap: bool,
) -> ::std::result::Result<MessageCallResult, TrapKind> {
    if _receive_address == Address::hex!("0x1") {
        let d = // do ecrecover here
        Ok(MessageCallResult::Success(*gas, ReturnData::&*d))
    }
}
```

**Comment by armaniferrante** [comment]
@ArjunBhuptani that seems like a good place to start. You also need to keep track of the callstack depth and implement the `depth` externality--though you can probably just return a fake constant since it's only used on [calls](https://github.com/openethereum/openethereum/blob/master/ethcore/evm/src/interpreter/mod.rs#L685).

**Comment by ArjunBhuptani** [comment]
Ah good tip, thanks!

**Comment by armaniferrante** [comment]
@ArjunBhuptani I updated the repo to use the latest openethereum packages, since everything moved over from paritytech to a new org. You'll want to pull the latest changes.

**Comment by LayneHaber** [comment]
Alright, so working on creating a test that will emulate this properly within our fork. Having a bit of trouble getting the regular node tests passing, are there specific instructions for running those?

**Comment by LayneHaber** [comment]
Right now the workflow is:
1. `cargo build` in root and in `/wasm`
2. `make build` in `/wasm`
3. `yarn install && yarn test` in `/examples/node`

but I seem to be messing things up along the way

**Comment by LayneHaber** [comment]
I seem to be getting this error when trying to merge the browser and node packages: 
```sed: pkg-node/pure-evm_bg.js: No such file or directory```

**Comment by LayneHaber** [comment]
Yeah, as soon as I use the imports from `../pkg-node` it starts to behave

**Comment by armaniferrante** [comment]
You should only need steps 2 and 3, as listed above.

I'm guessing you don't have `wasm-pack` installed, since `pkg-node` shouldn't be there if the build was successful?

To install, you can use the docker container with all the required deps or install it with this line https://github.com/armaniferrante/pure-evm/blob/master/.circleci/docker/Dockerfile#L18

**Comment by LayneHaber** [comment]
yeah there was a different error when I didn't have wasm installed. not sure what's off with my build step quite yet, but found a workaround so i'll probably come back to that later

**Comment by ArjunBhuptani** [comment]
PR: #13
